### PR TITLE
WIP: Angular 6 upgrade

### PR DESCRIPTION
**Pull request by pstjean**
This brings in the new versions of `ngx-prx-styleguide` and `prx-ng-serve`, as well as upgrading `rxjs` and all `@angular` packages to 6.1.10.

Tweaks to polyfills, build and various and sundry other things.

### CODE REVIEW

**Comment by svevang** [comment]
Love it! This is so great.
Ran though a `docker-compose build` and stepped through a podcast import. Looks good.
 I'm seeing a few missing assets:
screen shot 2019-02-14 at 1 31 08 pm
screen shot 2019-02-14 at 1 32 43 pm

**Comment by pstjean** [comment]
Good catch @svevang! Pushed a fix

**Comment by pstjean** [comment]
Discovered a couple of insidious, browser tab destroying bugs in the dev build, so holding for now.

File: angular.json [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            ],
+            "styles": [
+              "node_modules/pikaday/css/pikaday.css",
+              "node_modules/pikaday/css/triangle.css",
+              "node_modules/c3/c3.css",
+              "node_modules/ngx-prx-styleguide/assets/styles/bundle.scss",
+              "src/styles.css"
+            ],
+            "assets": [
+              "src/assets",
```
[the remark this answers is not in the document]
Comment (pstjean) [reply angular.json]: Yep, Sam caught those and I've pushed an update for that. Added docs in Styleguide as well.

File: package.json [hunk 2]
```diff
@@ -58,20 +57,23 @@
     "prosemirror-schema-list": "0.18.0",
     "prosemirror-state": "0.18.0",
     "prosemirror-view": "0.18.0",
-    "prx-ng-serve": "^0.0.4",
-    "rxjs": "^5.5.2",
+    "prx-ng-serve": "^1.0.0",
+    "rxjs": "~6.3.0",
+    "rxjs-compat": "^6.0.0-rc.0",
```
[the remark this answers is not in the document]
Comment (pstjean) [reply package.json]: Turns out there were a lot of bad imports in the test suite, but they're gone now!

File: src/app/series/series-import.service.ts [hunk 3]
```diff
@@ -18,17 +19,16 @@ export class SeriesImportService {
 
     let lastReceived = null;
 
-    let seriesImportPoller = Observable
-      .interval(1000)
-      .flatMap(() => {
+    let seriesImportPoller = interval(1000)
+      .pipe(flatMap(() => {
```
[the remark this answers is not in the document]
Comment (pstjean) [reply src/app/series/series-import.service.ts]: Fixed

File: src/app/story/directives/basic.component.css [hunk 4]
```diff
@@ -9,9 +9,7 @@ hr {
 }
 
 .file-select {
```
Comment (sandikbarr) [inline src/app/story/directives/basic.component.css]: Pretty sure this positioning was actually for the select to be positioned to the right above the file select upload buttons for uploading audio files... The select is below the label now. It does seem like a minor thing if this was a problem.

File: package.json [hunk 5]
```diff
@@ -23,32 +23,31 @@
...
+    "angulartics2": "^6.1.10",
     "codecov": "PRX/codecov-node",
     "core-js": "^2.4.1",
     "evaporate": "^2.1.4",
-    "intl": "^1.2.5",
     "langmap": "0.0.14",
     "moment": "^2.18.1",
     "ng2-dragula": "^1.3.1",
-    "ngx-prx-styleguide": "1.1.1",
+    "ngx-prx-styleguide": "^2.0.0",
```
Comment (sandikbarr) [inline package.json]: We've actually been pinning this to a specific version, which is currently "2.0.1"

Comment (pstjean) [reply package.json]: Fixed

File: src/app/story/directives/basic.component.css (hunk 4 above)
Comment (pstjean) [reply src/app/story/directives/basic.component.css]: The default styling for the new dropdown expands arbitrarily within its container to show all of the selected options. I removed the absolute positioning and moved it below the label to allow that behavior without covering up other elements.

**Pull request merged**